src/main/java/software/amazon/smithy/lsp/Utils.java

```diff
@@ -93,7 +87,50 @@ public static boolean isFile(String rawUri) {
     } finally {
       is.close();
     }
+
     return result;
   }
 
+  /**
+   * @param rawUri      the uri to a file in a jar.
+   * @param classLoader a classloader used to retrieve resources.
+   * @return the lines of the jar file, as a list.
+   * @throws IOException when rawUri cannot be URI-decoded.
+   */
+  public static List<String> jarFileContents(String rawUri, ClassLoader classLoader) throws IOException {
+    String uri = java.net.URLDecoder.decode(rawUri, StandardCharsets.UTF_8.name());
+    String[] pathArray = uri.split("!/");
+    String resourcePath = pathArray[1];
+    InputStream is = classLoader.getResourceAsStream(resourcePath);
+
+    return getLines(is);
+  }
+
+  /**
+   * @param rawUri      the uri to a file in a jar.
+   * @param classLoader a classloader used to retrieve resources.
+   * @return the lines of the jar file, as a list.
+   * @throws IOException when rawUri cannot be URI-decoded.
+   */
+  public static List<String> jarFileContents(String rawUri) throws IOException {
```

Comment:
A version of the method above to work with external jars, where we load files directly from jars.

I think messing with classloaders is 99% of the time a bad idea, so if we can keep it on file level - we have a lot more flexibility